Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -246,11 +282,11 @@ void AsyncScrollingCoordinator::frameViewRootLayerDidChange(LocalFrameView& fram
     
     // If the root layer does not have a ScrollingStateNode, then we should create one.
     ensureRootStateNodeForFrameView(frameView);
-    ASSERT(m_scrollingStateTree->stateNodeForID(frameView.scrollingNodeID()));
+    ASSERT(ensureScrollingStateTreeForRootFrameID(frameView.frame().rootFrame().frameID()).stateNodeForID(frameView.scrollingNodeID()));
 
     ScrollingCoordinator::frameViewRootLayerDidChange(frameView);
 
-    RefPtr node = dynamicDowncast<ScrollingStateFrameScrollingNode>(m_scrollingStateTree->stateNodeForID(frameView.scrollingNodeID()));
+    RefPtr node = dynamicDowncast<ScrollingStateFrameScrollingNode>(ensureScrollingStateTreeForRootFrameID(frameView.frame().rootFrame().frameID()).stateNodeForID(frameView.scrollingNodeID()));
```

Comment:
Avoid long call chains; pull the `ensureScrollingStateTreeForRootFrameID()` out onto a previous line.

Reply:
Fixed.